addons/humor/yeetgun/fn_yeetGunFired.sqf

```diff
@@ -0,0 +1,98 @@
+params ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_gunner"];
+private _barrelPos = eyePos _unit;
+if (local _unit) then {
+    // Shitty hack for 0 damage bullets, as AI refuse to shoot 0 damage ammo.
+    deleteVehicle _projectile;
+};
+private _range = 85;
+private _pressureBase = 13500; // Wind pressure of 335mph wind
+private _barrelFOV = 45;
+private _nearby = nearestObjects [_unit, ["Static", "CAManBase", "AllVehicles"], _range] - [_unit];
+private _weaponDirVec = _unit weaponDirection currentWeapon _unit;
+{
+    private _target = _x;
+    if (local _target) then {
+        private _vectorDirTo = _barrelPos vectorFromTo getPosWorld _target;
+        private _distanceToTarget = _target distance _unit;
+        private _distanceCoef = 0 max (1 - _distanceToTarget/_range);
+        private _weaponDirTo = (_vectorDirTo vectorDiff _weaponDirVec);
+        private _weaponDirToMagnitude  = vectorMagnitude _weaponDirTo;
+        private _effectiveAngle = deg pi*_weaponDirToMagnitude/2;
+        private _angle = _barrelFOV min _effectiveAngle;
+        private _angleCoef  = 1 - (_angle/_barrelFOV);
+        if (_distanceCoef == 0 || _angleCoef == 0) then {continue};
+        
+        // Visiblity check
+        private _bbr = 0 boundingBoxReal _target;
+        private _p1 = _bbr select 0;
+        private _p2 = _bbr select 1;
+        private _canHit = false;
+        // Checks the 8 bounding box points to see if any part of the object is visible. Horrible, right?
+        for "_i" from 0 to 1 do {
+            private _posX = [_p1#0, _p2#0]#_i;
+            for "_j" from 0 to 1 do {
+                private _posY = [_p1#1, _p2#1]#_j;
+                for "_k" from 0 to 1 do {
+                    private _posZ = [_p1#2, _p2#2]#_k;
+                    private _pos = AGLtoASL (_target modelToWorld [_posX, _posY, _posZ]);
+                    if !(lineIntersects [_barrelPos, _pos, _unit, _target]) exitWith {_canHit = true}
+                };
+                if (_canHit) exitWith {}
+            };
+            if (_canHit) exitWith {};
+        };
+        if !(_canHit) then {continue};
+
+        private _case = ["Static", "CAManBase", "AllVehicles"] findIf {_target isKindOf _x};
```

Comment:
If you want to make this slightly clearer, you can actually give expressions to `case` in SQF and, in this case, they would have the same performance impact.

```sqf
switch true do {
    case (_target isKindOf "Static"): {
        // Terrain object ...
    };
    case (_target isKindOf "CAManBase"): {
        private _targetArea = 0.75;
         ...
    };
    ...
};
```
https://community.bistudio.com/wiki/switch_do